src/core/src/pass/serialize.cpp

```diff
@@ -1229,21 +1229,27 @@ bool pass::Serialize::run_on_model(const std::shared_ptr<ov::Model>& model) {
         if (xmlDir != m_xmlPath)
             ov::util::create_directory_recursive(xmlDir);
 
-        std::ofstream bin_file(m_binPath, std::ios::out | std::ios::binary);
-        OPENVINO_ASSERT(bin_file, "Can't open bin file: \"" + m_binPath + "\"");
-
-        // create xml file
-        std::ofstream xml_file(m_xmlPath, std::ios::out);
-        OPENVINO_ASSERT(xml_file, "Can't open xml file: \"" + m_xmlPath + "\"");
-
         try {
+            // create bin file
+            std::ofstream bin_file(m_binPath, std::ios::out | std::ios::binary);
+            bin_file.exceptions(std::ofstream::failbit | std::ofstream::badbit);
+            OPENVINO_ASSERT(bin_file, "Can't open bin file: \"" + m_binPath + "\"");
+        
+            // create xml file 
+            std::ofstream xml_file(m_xmlPath, std::ios::out);
+            xml_file.exceptions(std::ofstream::failbit | std::ofstream::badbit);
+            OPENVINO_ASSERT(xml_file, "Can't open xml file: \"" + m_xmlPath + "\"");
+
             serializeFunc(xml_file, bin_file, model, m_version);
-        } catch (const ov::AssertFailure&) {
-            // optimization decision was made to create .bin file upfront and
-            // write to it directly instead of buffering its content in memory,
-            // hence we need to delete it here in case of failure
-            xml_file.close();
-            bin_file.close();
+        } catch (const std::ofstream::failure& e) {
```

Comment:
is it possible to cover such scenarios with tests?

Reply:
I suggest not catching `std::ofstream::failure` because it would be more expected for the user that file exists even if it isn't ready to use (i.e. leave a responsibility for the user to clean file system)